### DISCUSSION TITLE
Preferred way to update a merged kata translation

### DISCUSSION THREAD

**Discussion by FranzMartyn** [codewars/codewars.com#3269] [Q&A] [answered]
I had and may still have to update some translations that have been merged, either due to using different functions for testing, or to just simply improve it. 

What is the preferred way of doing so? 

1. Fork the old kumite and do the changes in the fork
2. Update the kumite directly
3. Something else???

**Comment by hobovsky** [comment, marked as answer]
If you want to change the existing, current state of a working kata, you have two options:
- Preferred: use "fork" button **on the kata page** (in the top right corner). This will create new kumite, where you introduce your changes, put small summary of changes in its initial comment, publish it, advertise your fork in kata discourse, and wait till it gets approved.
- Less preferred but also possible: use kata editor to directly edit the kata. However this way is not the great workflow, skips some useful steps, requires high privileges, makes you forfeit all the future progress from the kata, and that's why the former way is preferred.
- Bad way to go about this is to reuse some old kumite (pending or finalized in some way), fork it, and publish it. The older such kumite, the more outdated it is, and merging it can be difficult or give undesired result.

If you want to improve a kumite because for example it is a fix or a new translation going through rounds of reviews:
- If the kumite is still pending (i.e. not approved and not rejected), _and_ if it has no important descendants (i.e. is a leaf kumite or its descendants are discardable), _and_ it did not get outdated (i.e. kata description in the kumite is identical the current kata description and did not change since the kumite was created), _and_ you are the author of the kumite, you can edit it and republish it with changes. This is usually the workflow when iteratively going though reviews of a new, fresh translation and applying suggestions and remarks from reviews.
- In any other case (i.e. if it's not yours, or if it's outdated, or if it got approved in the meantime, or it's an old, stale, unapproved translation), you need to fork the kumite and publish the fork with your changes. This requires a bit of attention, and it's important to make sure that it does not use outdated description. Potentially important thing to note is that when such fork gets approved, all credit goes to the **author of the top level kumite**.
- It is also possible to reject the currently pending kumite and create a new one. But it's recommended only for severely outdated or bad ones, because it strips the original author of the top level kumite from all the credit.

If you need some guidance how to proceed with some translation, you can always ask for help in the `#help-translate` channel of Codewars Discord.
